streams/src/test/java/org/apache/kafka/streams/integration/AbstractJoinIntegrationTest.java

```diff
@@ -166,35 +138,12 @@ void prepareEnvironment() throws InterruptedException {
         }
 
         STREAMS_CONFIG.put(StreamsConfig.STATE_DIR_CONFIG, testFolder.getRoot().getPath());
-
-        producer = new KafkaProducer<>(PRODUCER_CONFIG);
     }
-
     @After
     public void cleanup() throws InterruptedException {
-        producer.close(Duration.ofMillis(0));
         CLUSTER.deleteAllTopicsAndWait(120000);
```

Comment:
This is still needed.  We have a couple of tests (non-processing) that start a Kafka Streams application and need the topics to exist, hence we delete them here.

Reply:
Q: Does this mean, we still start an embedded Kafka before each test? Is it possible to start an embedded Kafka only in the tests that need it?

Reply:
+1